language/functional_tests/tests/testsuite/validator_set/reconfiguration.mvir

```diff
@@ -0,0 +1,125 @@
+//! account: alice
+//! account: vivian, 1000000, 0, validator
+//! account: viola, 1000000, 0, validator
+
+//! new-transaction
+//! sender: alice
+import 0x0.ValidatorConfig;
+
+// register Alice as a validator candidate
+main() {
+    let config: ValidatorConfig.Config;
+
+    ValidatorConfig.register_candidate_validator(h"10", h"20", h"30");
+    return;
+}
+
+// check: EXECUTED
+
+//! new-transaction
+//! sender: alice
+import 0x0.LibraSystem;
+import 0x0.ValidatorConfig;
+
+// rotate alice's pubkey and then run the block prologue. No reconfiguration should be triggered,
+// since alice is not a validator
+// TODO: trigger a BlockMetadata transaction here instead
+main() {
+    ValidatorConfig.rotate_consensus_pubkey(h"40");
+    LibraSystem.block_prologue(1);
+
+    return;
+}
+
+// check: events: []
+// check: EXECUTED
+
+//! new-transaction
+//! sender: vivian
+import 0x0.LibraSystem;
+import 0x0.ValidatorConfig;
+
+// rotate vivian's pubkey and then run the block prologue. Now, reconfiguration should be triggered.
+main() {
+    let info: LibraSystem.ValidatorInfo;
+
+    ValidatorConfig.rotate_consensus_pubkey(h"40");
+    LibraSystem.block_prologue(2);
```

Comment:
might be nice to verify that the old key is still in use prior to the block_prologue